clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/IngestionBatchProcessor.java

```diff
@@ -99,12 +102,12 @@ public Map<ByteArrayKey, ReentrantLock> lockKeys(List<PubSubMessage<KafkaKey, Ka
       keyLockMap.forEach((k, v) -> v.lock());
       return keyLockMap;
     }
-    return Collections.emptyMap();
+    return EMPTY_TREE_MAP;
```

Comment:
Instead of this, can you please make the `lockKeys` function return a `NavigableMap` and return `Collections.emptyNavigableMap()` here? The issue with the current code is that the statically-defined `EMPTY_TREE_MAP` is modifiable, which is not safe... (of course, the current code never modifies it, but it would be more defensive to make it unmodifiable). `unlockKeys` and whatever else function gets passed a `TreeMap` would need to be switched to `NavigableMap` as well.

Reply:
Fixed.